tests/CommunityToolkit.Aspire.Hosting.EventStore.Tests/AppHostTests.cs

```diff
@@ -0,0 +1,109 @@
+{
+    [Fact]
+    public async Task ResourceStartsAndRespondsOk()
+    {
+        var resourceName = "eventstore";
+        await fixture.ResourceNotificationService
+            .WaitForResourceAsync(resourceName, KnownResourceStates.Running)
+            .WaitAsync(TimeSpan.FromMinutes(1));
+
+        var httpClient = fixture.CreateHttpClient(resourceName);
+
+        var response = await httpClient.GetAsync("/");
+
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+    [Fact]
+    public async Task ApiServiceCreateAccount()
+    {
+        var resourceName = "apiservice";
+        await fixture.ResourceNotificationService
+            .WaitForResourceAsync(resourceName, KnownResourceStates.Running)
+            .WaitAsync(TimeSpan.FromMinutes(1));
+
+        var httpClient = fixture.CreateHttpClient(resourceName);
+
+        var createResponse = await httpClient.PostAsJsonAsync("/account/create", new { });
+        createResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var location = createResponse.Headers.Location;
+
+        var getResponse = await httpClient.GetAsync(location);
+        getResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var account = await getResponse.Content.ReadFromJsonAsync<AccountDto>();
+        Assert.NotNull(account);
+        Assert.Equal("John Doe", account.Name);
+        Assert.Equal(100, account.Balance);
+    }
+
+    [Fact]
+    public async Task ApiServiceCreateAccountAndDeposit()
+    {
+        var resourceName = "apiservice";
+        await fixture.ResourceNotificationService
+            .WaitForResourceAsync(resourceName, KnownResourceStates.Running)
+            .WaitAsync(TimeSpan.FromMinutes(1));
+
+        var httpClient = fixture.CreateHttpClient(resourceName);
+
+        var createResponse = await httpClient.PostAsJsonAsync("/account/create", new { });
+        createResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var location = createResponse.Headers.Location;
+
+        var depositResponse = await httpClient.PostAsJsonAsync($"{location!}/deposit", new { Amount = 50 });
+        depositResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var getResponse = await httpClient.GetAsync(location);
+        getResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+
+        var account = await getResponse.Content.ReadFromJsonAsync<AccountDto>();
+        Assert.NotNull(account);
+        Assert.Equal("John Doe", account.Name);
+        Assert.Equal(150, account.Balance);
+    }
+
+    [Fact]
+    public async Task ApiServiceCreateAccountAndWithdraw()
+    {
+        var resourceName = "apiservice";
+        await fixture.ResourceNotificationService
+            .WaitForResourceAsync(resourceName, KnownResourceStates.Running)
```

Comment:
it's better to use the method that waits for the resource to be healthy, so consideration is taken for the app in the container starting up

Reply:
Addressed in the last commit: https://github.com/CommunityToolkit/Aspire/pull/277/commits/7deb7d5bc32aa830d8266342076c010509f75b88